CHANGELOG.md

```diff
@@ -2,6 +2,14 @@
 
 Please, check out guidelines: https://keepachangelog.com/en/1.0.0/
 
+## 2.0
+
+### Removed
+
+- **Breaking** `.cocoapods` target dependency
+  - **Motivation:** `.cocoapods`'s API led users to believe their integration issues were Tuist's fault. Therefore we decided to remove it and make it an explicit action developers need to run after the generation of Xcode projects through Tuist.
+  - **Migration:** we recommend wrapping the the generation of projects in a script that runs `pod install` right after generating the project: `tuist generate && pod install`. Alternatively, you might consider adopting Swift Package Manager and using our built-in support for package dependencies through the `Dependencies.swift` manifes tfile.
```

Comment:
I think we should include this comment in the documentation - ideally, [here](https://github.com/tuist/tuist/pull/3369/files?file-filters%5B%5D=.md#diff-4444ec2df9ccefa937f2f7dfdd2b1b82e36f5ca1bfea3ba387c73750912a84b1L103) where you removed the interaction with the Cocoapods API. This will be probably asked by the users (how I do integrate tuist + Cocoapods?) and to mitigate these questions in the first place, it's probably a good idea to have this in the documentation.

Going forward, I'd really like to work on post-generation tasks hooks but for now, we can delegate this responsibility to the teams.

Reply:
Good idea. I pushed a commit adding documentation explaining why we dropped that API in 2.0. I like the idea of having hooks, but it requires further thinking and a proposal to understand all the implications of having such API.